### PR TITLE
DATAUP-600 results tab crashes with deleted files

### PR DESCRIPTION
**Pull request by briehl**
# Description of PR purpose/changes

The results tab was crashing on a later load if any of the generated files were deleted in the meantime. This handles those errors in a somewhat more elegant way and shows some placeholder text for missing data.

This screenshot demonstrates the various missing data states:
Screen Shot 2021-10-29 at 2 09 00 PM

Some of them, in a few rare cases, are recoverable - if there's no object name given in the results, the name from the fetched workspace object info can be used. Same for the type. However, these are also fetched from the workspace, so if there's an error with one ws call, there's likely an error with others, so this is kind of an odd edge case.

For the reports area, if the report itself isn't found, I left a row for it, and just put some dummy text with "report not found" next to it. These will only show up if there's expected to be a report associated with an object, but the viewer gets a null - if no report is expected anyway, then nothing should show up. But all of our uploaders have reports, so that's how this will work for now.

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-600
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
* Run a bulk import cell, delete some or all of the generated data, and load up the results.
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/results/outputWidget.js [hunk 1]
```diff
@@ -51,23 +54,39 @@ define([
...
+                                ? obj.wsInfo[1]
+                                : 'Unknown object name';
+                            const type = obj.type
+                                ? obj.type
+                                : obj.wsInfo
+                                ? obj.wsInfo[2]
+                                : 'Missing type';
+                            const parsedType = APIUtil.parseWorkspaceType(type) || { type };
+                            const description = obj.description || 'Missing description';
+                            let objLink = '';
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/results/outputWidget.js:69]: you could initialise it to `name`, and then change it if `obj.wsInfo` is available -- saves a couple of lines.

File: kbase-extension/static/kbase/js/common/cellComponents/tabs/results/reportWidget.js [hunk 2]
```diff
@@ -31,39 +31,61 @@ define(['bluebird', 'jquery', 'common/html', 'common/ui', 'common/events', 'kbas
...
+                                handler: (e) => toggleReportView(e, objInfo.reportRef),
                             }),
                         },
-                        objInfo.name
-                    ),
-                ]);
+                        name
+                    );
+                } else {
+                    reportElem = name + ' (report not found)';
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/cellComponents/tabs/results/reportWidget.js:59]: again, could init reportElem with this value and change it if `objInfo.reportRef` is available

**Review by ialarmedalien — APPROVED** [review]
A couple of minor suggestions, otherwise looks great!

**Comment by briehl** [comment]
@ialarmedalien Thanks! I took your suggestions, and put an update that removes the nested ternary statements. Because (1) prettier couldn't handle them, and (2) ew.

**Comment by ialarmedalien** [comment]
Merge whenever the build is complete!

**Pull request merged**